### PR TITLE
Fix some warnings and avoid false-positive issues

### PR DESCRIPTION
**Pull request by dreamer**
Please review commit-by-commit.

### CODE REVIEW

File: src/misc/setup.cpp [hunk 1]
```diff
@@ -852,15 +852,16 @@ void Config::Init() {
...
+{
+	initfunctions.emplace_back(func, canchange);
 }
 
-void Section::AddDestroyFunction(SectionFunction func,bool canchange) {
-	destroyfunctions.push_front(Function_wrapper(func,canchange));
+void Section::AddDestroyFunction(SectionFunction func, bool canchange)
+{
+	destroyfunctions.emplace_back(func, canchange);
 }
```
Comment (kcgen) [inline src/misc/setup.cpp:863]: Maybe `emplace_front`, if we want to strictly have the same functionality? (although maybe it doesn't matter).

Comment (dreamer) [reply src/misc/setup.cpp:863]: Good catch. It probably does not matter, but I didn't want to change the behaviour.

Comment (dreamer) [reply src/misc/setup.cpp]: Ops, I almost introduced a bug by changing both to `emplace_front`.

This actually does matter: we want respective destroy functions to be run in reverse order compared to init functions after all. Dodged a bullet in here :)

Comment (kcgen) [reply src/misc/setup.cpp]: phew :sweat_smile:

**Pull request merged**